pandas-stubs/core/indexes/interval.pyi

```diff
@@ -19,38 +24,46 @@ class IntervalIndex(IntervalMixin, ExtensionIndex):
     def __new__(
         cls,
         data,
-        closed=...,
-        dtype=...,
+        closed: Literal["left", "right", "both", "neither"] = ...,
```

Comment:
Could use `IntervalClosedType` from `pandas._typing`

Reply:
And elsewhere in this PR as well to use `IntervalClosedType`